ui/apps/dashboard/src/components/Billing/Addons/EntitlementListItem.tsx

```diff
@@ -0,0 +1,137 @@
+'use client';
+
+import { Button } from '@inngest/components/Button';
+import { Tooltip, TooltipContent, TooltipTrigger } from '@inngest/components/Tooltip/Tooltip';
+import { RiInformationLine } from '@remixicon/react';
+
+import EntitlementListItemSelfService from '@/components/Billing/Addons/EntitlementListItemSelfService';
+import { addonPriceStr } from '@/components/Billing/Addons/pricing_help';
+import { pathCreator } from '@/utils/urls';
+
+export default function EntitlementListItem({
+  title,
+  description,
+  tooltipContent,
+  selfServiceAvailable,
+  canIncreaseLimitInCurrentPlan,
+  entitlement,
+  addon,
+  onChange,
+}: {
+  title: string; // Title of the addon (e.g. Users, Concurrency)
+  description?: string;
+  tooltipContent?: string | React.ReactNode;
+  selfServiceAvailable: boolean; // Is an addon available (for this account) via self-service or will they need to contact us?
+  canIncreaseLimitInCurrentPlan: boolean; // Can this limit be increased on the current billing plan (or does the user need to upgrade to a higher plan)?
+  entitlement: {
+    currentValue?: number | boolean; // Current value of the entitlement relevant to this addon, including plan + any addons + account-level overrides
+    displayValue?: string;
+    planLimit?: number; // The amount of this addon included in the current billing plan
+    maxValue?: number; // The maximum amount of (value * quantityPer) that can be purchased
+  };
+  addon?: {
+    addonName: string;
+    quantityPer: number; // The number of units (e.g. concurrency or users) included in one purchase of this addon
+    price: number; // Price for one purchase of this addon, in US Cents
+  };
+  onChange?: () => void;
+}) {
+  const isPlanEntUnlimited = entitlement.planLimit === undefined || entitlement.planLimit === -1;
```

Comment:
The UI doesn't need to handle the `-1` case. Our resolvers convert `-1` to `null`, e.g.:
https://github.com/inngest/monorepo/blob/VercelMarketAPI-login/api/gql/resolver/entitlements.go#L206-L209

The backend needs to differentiate between explicitly unlimited (`-1`) and implicitly unlimited (not set), but the UI doesn't. The UI just needs to know if it's unlimited